cpp/benchmarks/profiling.md

```diff
@@ -1,4 +1,39 @@
-# Performance measuring with Score-P
+# Profiling
+
+This document explains how to create performance profiles, i.e., measurements of where in the code time is spent, usually aggregated by function. There is a variety of open-source or proprietary tools available. Generally, there are three types of profilers:
+- instrumenting: modify the compiled code to include code for measurement. Advantage: exact function call counts and times, no approximations. Disadvantage: instrumentation adds runtime overhead that can distort the result or disable compiler optimizations (e.g. functions aren't inlined), so short functions usually must be excluded.
+- sampling: checking the CPU stack register in regular intervals to see which code is currently being executed. Advantage: usually low measurement runtime overhead. Disadvantage: stochastic results, imprecise especially for short functions, longer runtimes improve averages.
+- emulating: running the program in a virtual environment. Advantage: exact and able to measure even small functions. Disadvantage: program can run significantly longer, no real world timing results but instruction cycle counts are generally directly translatable.
+
+These tools have been tried for memilio and are described below:
+- gperftools: sampling profiler that gives reliable results with little setup
+- Score-P: large suite that supports both sampling and instrumenting. Special support for parallel code (OpenMP and MPI), including tracing to find load imbalances in the parallelization. 
+- Valgrind: suite that includes the Callgrind emulating profiler; description NYI; see [here](https://valgrind.org/info/tools.html)
```

Comment:
I don't like the abbreviation NYI here (I at least don't see it used often), how about 

```suggestion
- Valgrind: suite that includes the Callgrind emulating profiler; a description can be found at the [official site](https://valgrind.org/info/tools.html)
```

or something similar. Also, using callgrind is pretty easy, if you like we could add the steps:
1. Compile with mode "RelWithDebInfo".
2. Run `valgrind --tool=callgrind \<path/to/binary\>`
3. Open "callgrind.out.\<PID\>" in KCacheGrind to visualize the call graph.

Reply:
we are working on a full explanation. You are right that the wording was weird, improved it.